src/Microsoft.AspNet.Authentication/Notifications/RedirectToIdentityProviderNotification.cs

```diff
@@ -1,16 +1,64 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Authentication;
+using Microsoft.Framework.Internal;
 
 namespace Microsoft.AspNet.Authentication.Notifications
 {
+    /// <summary>
+    /// When a user configures the <see cref="AuthenticationMiddleware{TOptions}"/> to be notified prior to redirecting to an IdentityProvider
+    /// and instance of <see cref="RedirectFromIdentityProviderNotification{TMessage, TOptions, TMessage, AuthenticationProperties}"/> is passed to 'RedirectToIdentityProvider".
+    /// </summary>
+    /// <typeparam name="TMessage">protocol specific message.</typeparam>
+    /// <typeparam name="TOptions">protocol specific options.</typeparam>
     public class RedirectToIdentityProviderNotification<TMessage, TOptions> : BaseNotification<TOptions>
     {
-        public RedirectToIdentityProviderNotification(HttpContext context, TOptions options) : base(context, options)
+        TMessage _message;
+        AuthenticationProperties _properties;
+
+        public RedirectToIdentityProviderNotification([NotNull] HttpContext context, [NotNull] TOptions options, [NotNull] TMessage protocolMessage, [NotNull] AuthenticationProperties properties ) : base(context, options)
+        {
+            ProtocolMessage = protocolMessage;
+            AuthenticationProperties = properties;
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="{TMessage}"/>.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">if 'value' is null.</exception>
+        public TMessage ProtocolMessage
         {
+            get { return _message; }
```

Comment:
Is the public set really necessary on Message/AuthProperties?  can these not be private and only settable via the ctor?  If the sets are only needed for unit internal set, rather than adding null checks seems better.